src/app/motor-inventory/motor-inventory-summary/inventory-summary-graphs/inventory-summary-graphs-menu/inventory-summary-graphs-menu.component.ts

```diff
@@ -0,0 +1,117 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { MotorInventoryService } from '../../../motor-inventory.service';
+import { InventorySummaryGraphsService } from '../inventory-summary-graphs.service';
+import { Subscription } from 'rxjs';
+import { InventorySummaryTableService } from '../../inventory-summary-table/inventory-summary-table.service';
+import { MotorInventoryData } from '../../../motor-inventory';
+import { Settings } from '../../../../shared/models/settings';
+import * as _ from 'lodash';
+@Component({
+  selector: 'app-inventory-summary-graphs-menu',
+  templateUrl: './inventory-summary-graphs-menu.component.html',
+  styleUrls: ['./inventory-summary-graphs-menu.component.css']
+})
+export class InventorySummaryGraphsMenuComponent implements OnInit {
+  @Input()
+  settings: Settings;
+
+  selectedFieldSub: Subscription;
+  selectedField: { display: string, value: string, group: string };
+  graphTypeSub: Subscription;
+  graphType: string;
+  groups: Array<{
+    options: Array<{ display: string, value: string, group: string }>,
+    groupLabel: string,
+    showGroup: boolean
+  }>
+  constructor(private motorInventoryService: MotorInventoryService, private inventorySummaryGraphService: InventorySummaryGraphsService,
+    private inventorySummaryTableService: InventorySummaryTableService) { }
+
+  ngOnInit(): void {
+    this.setOptions();
+    this.selectedFieldSub = this.inventorySummaryGraphService.selectedField.subscribe(val => {
+      if (val) {
+        this.selectedField = val;
+      } else {
+        let ratedPowerOption = _.find(this.groups[0].options, (option) => { return option.value == 'ratedMotorPower' });
```

Comment:
Wondering what lodash has here over native .find(). Could drop the lodash import otherwise?

Reply:
Lodash is more performative and it's a large array